packages/dataclass-editor/src/protocol/types.ts

```diff
@@ -4,6 +4,7 @@ export type Data = { context: DataContext; data: DataClass };
 export type DataContext = { app: string; pmv: string; file: string };
 export type EditorProps = { context: DataContext; directSave?: boolean };
 export type SaveArgs = Data & { directSave?: boolean };
+export type CombineFieldsContext = { context: DataContext; fieldNames: Array<string> };
```

Comment:
I assume a name for the new data class also needs to be provided by the user in some way (dialog?) and sent to the backend. But you'll get there when you implement this further.

Reply:
Currently the combine will only work in the eclipse world. and there is already a SWT based dialog for this.